### PR TITLE
feat(datasource-customizer): add support for binary fields

### PR DESCRIPTION
**Pull request by romain-gilliotte**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/_example/src/forest/typings.ts [hunk 1]
```diff
@@ -44,6 +45,7 @@ export type Schema = {
       'parent:firstname': string;
       'parent:lastname': string;
       'parent:storeId': number;
+      'parent:avatar': Buffer;
```
Comment (ShohanRahman) [inline packages/_example/src/forest/typings.ts:48]: all types in the files are in lowercase, it is not following the pattern, is there a reason ? where `typing` file is interpreted ?

File: packages/datasource-customizer/src/collection-customizer.ts [hunk 2]
```diff
@@ -421,6 +422,25 @@ export default class CollectionCustomizer<
...
+   * short binary data (for instance binary uuids).
+   *
+   * @param name the name of the field
+   * @param binaryMode either 'datauri' or 'hex'
+   * @example
+   * .replaceFieldBinaryMode('avatar', 'datauri');
+   */
+  replaceFieldBinaryMode(name: TColumnName<S, N>, binaryMode: BinaryMode): this {
+    return this.pushCustomization(async () => {
+      this.stack.binary.getCollection(this.name).setBinaryMode(name, binaryMode);
```
Comment (ShohanRahman) [inline packages/datasource-customizer/src/collection-customizer.ts:440]: `getCollection` is used to retrieve a column ? isn't the naming not correct ? 
I interpreted it as a getTable method but we are handling **column name**

Comment (romain-gilliotte) [reply packages/datasource-customizer/src/collection-customizer.ts:440]: `this.name` is the collection name
`name` is the column name

File: packages/_example/src/forest/typings.ts (hunk 1 above)
Comment (romain-gilliotte) [reply packages/_example/src/forest/typings.ts:48]: This file is autogenerated from the structure of the database of the example agent.
It's the file that enables autocompletion in the IDE of the customers

It is documented here: https://docs.forestadmin.com/developer-guide-agents-nodejs/getting-started/install/autocompletion-and-typings

Comment (ShohanRahman) [reply packages/_example/src/forest/typings.ts:48]: Thanks for the link. 
It remains the fact that the coherency is a bit odd with only this type in Pascal case

Comment (romain-gilliotte) [reply packages/_example/src/forest/typings.ts:48]: Ooh, you mean the `Buffer` versus `number`, `string`, etc...?

That's just how they are named, I have no control over it!
Those are native types from nodejs!

File: packages/datasource-customizer/src/decorators/binary/collection.ts [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    return value;
+  }
+
+  private async convertScalar(
+    toBackend: boolean,
+    useHex: boolean,
+    value: unknown,
+  ): Promise<unknown> {
+    if (toBackend) {
+      // direction === true
```
Comment (ShohanRahman) [inline packages/datasource-customizer/src/decorators/binary/collection.ts:233]: que veut dire ce com ?

File: packages/datasource-customizer/src/decorators/binary/collection.ts [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+      const string = value as string;
```
Comment (ShohanRahman) [inline packages/datasource-customizer/src/decorators/binary/collection.ts:234]: why using a temporary variable ?

Comment (romain-gilliotte) [reply packages/datasource-customizer/src/decorators/binary/collection.ts:234]: avoid duplicate the cast

File: packages/datasource-customizer/src/decorators/binary/collection.ts (hunk 3 above)
Comment (romain-gilliotte) [reply packages/datasource-customizer/src/decorators/binary/collection.ts:233]: Forgot it after refactoring the code. I removed it.

File: packages/datasource-customizer/src/collection-customizer.ts [hunk 5]
```diff
@@ -421,6 +422,25 @@ export default class CollectionCustomizer<
...
+   *
+   * Using 'datauri' allows to use the FilePicker widget, while 'hex' is more suitable for
+   * short binary data (for instance binary uuids).
+   *
+   * @param name the name of the field
+   * @param binaryMode either 'datauri' or 'hex'
+   * @example
+   * .replaceFieldBinaryMode('avatar', 'datauri');
+   */
+  replaceFieldBinaryMode(name: TColumnName<S, N>, binaryMode: BinaryMode): this {
```
[the remark this answers is not in the document]
Comment (romain-gilliotte) [reply packages/datasource-customizer/src/collection-customizer.ts:438]: added a ticket

**Review by ShohanRahman — APPROVED** [review]

**Pull request merged**